app/sessions.go

```diff
@@ -0,0 +1,53 @@
+package app
+
+import (
+	"database/sql"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type Session struct {
+	UserId    int
+	SessionId int
+	ExpiresAt time.Time
+}
+
+type SessionService struct {
+	db *sql.DB
```

Comment:
Again, the only purpose of us using the Repository pattern is to avoid having database code in our app logic. Create a `SessionRepository` interface to have methods for creating sessions (finding them by user ID, etc) on the persistence level. The `SessionService` should be responsible for calling these functions to manage the sessions properly. For example, your handler would call a method on `SessionService` to check if a session is valid based on the session ID from the token and you could potentially return the User if there is such a session, etc. 

Just remember that we want to avoid mixing server and database code into the app logic as much as we can.